chain/chain/src/chain.rs

```diff
@@ -3186,15 +3186,14 @@ impl Chain {
             store_update.commit()?;
         }
 
-        match self.runtime_adapter.try_create_flat_storage_state_for_shard(
-            shard_id,
-            block_height,
-            self.store(),
-        ) {
-            FlatStorageStateStatus::Ready | FlatStorageStateStatus::DontCreate => {}
-            status @ _ => {
-                return Err(Error::StorageError(StorageError::FlatStorageError(format!("Unable to create flat storage during syncing shard {shard_id}, got status {status:?}"))));
-            }
+        if self.runtime_adapter.get_flat_storage_creation_status(shard_id)
+            != FlatStorageCreationStatus::DontCreate
```

Comment:
I think the logic here is changed? Should this comparison be 
```
if self.runtime_adapter.get_flat_storage_creation_status(shard_id) == FlatStorageCreationStatus:Ready
```
instead?

Reply:
Well, right. By this I've tried to handle the case when runtime is `KeyValueRuntime. Anyway, it has to be done differently.